controllers/PollResponseController.ts

```diff
@@ -0,0 +1,162 @@
+     * @param {Request} req Represents request from client, including body
+     * containing the JSON object for the new pollResponse to be inserted in the
+     * database, if the Poll is close, then the response can't be created.
+     * @param {Response} res Represents response to client, including the
+     * body formatted as JSON containing the new pollResponse that was inserted in the
+     * database
+     */
+    createPollResponse = async (req: Request, res: Response) => {
+        // @ts-ignore
+        let userId = req.params.uid === "my" && req.session['profile'] ? req.session['profile']._id : req.params.uid;
+        // let userId = req.session['profile']._id;
+
+        let isPollClosed = await PollResponseController.pollResponseDao.isPollClosed(req.params.pid)
+        if(isPollClosed === true){
+            //The client shouldn't request this again
+            res.sendStatus(400);
+        //    the poll is not exist at all
+        }else if (isPollClosed === null){
+            res.sendStatus(404);
+        }
+        else{
+        PollResponseController.pollResponseDao.createPollResponse(userId,req.params.pid, req.body)
+                    .then((pollResponse: PollResponse) => res.json(pollResponse));
+        }
+
+    }
+
+    /**
+     * @param {Request} req Represents request from client, including path
+     * parameter tid identifying the primary key of the pollResponse to be modified
+     * if the poll is already closed, this will send a 404 code.
+     * @param {Response} res Represents response to client, including status
+     * on whether updating a pollResponse was successful or not
+     */
+    updatePollResponse = async (req: Request, res: Response) =>{
+        try {
+            let isPollClosed = await PollResponseController.pollResponseDao.isPollClosed(req.params.pid).catch(function(err) {
+                    console.error(err);
+                    res.status(500).json(err);})
+            if(isPollClosed === true){
+                console.log("Poll is closed");
+                res.sendStatus(400);
+            }else if (isPollClosed === null){
+                res.sendStatus(404);
+            }else{
+                PollResponseController.pollResponseDao.updatePollResponse(req.params.uid,req.params.pid,req.body)
+                    .then(status => res.json(status));
+            }
+        }catch (e) {
+            res.sendStatus(404)
+        }
+
+    }
+    /**
+     * @param {Request} req Represents request from client, including path
+     * parameter pid and uid representing pollId and userId, if the poll is already closed,
+     * the deletePollResponse will send a 404 code.
+     *  @param {Response} res Represents response to client, including status
+     *  on whether deleting a pollResponse was successful or not
+     */
+    deletePollResponse = async (req: Request, res: Response) =>{
+        let isPollClosed = await PollResponseController.pollResponseDao.isPollClosed(req.params.pid)
```

Comment:
We don't need a try catch block her like the one added for updateResponse function? If tested with incorrect pollID and it works then ignore this comment. If not tested please test

Reply:
delete it, but add isPollClosed DAO catch error